pkg/promclient/promclient.go

```diff
@@ -68,17 +71,36 @@ type Client struct {
 	HTTPClient
 	userAgent string
 	logger    log.Logger
+
+	metrics *clientMetrics
+}
+
+type clientMetrics struct {
+	requestCount *prometheus.SummaryVec
+}
+
+func newClientMetrics(reg prometheus.Registerer) *clientMetrics {
+	var m clientMetrics
+
+	m.requestCount = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
```

Comment:
We already have HTTP client metrics middleware for it. Let me double check if we don't instrument this right now...

Reply:
Actually we don't and we only have server HTTP middleware (and both client and server for GRPC), so this what we miss and we need to add :https://github.com/thanos-io/thanos/blob/8f0afeea960bd9c162edaa29642892716c396976/pkg/extprom/http/instrument_server.go#L35